Docs: Following `Appendix: Optimizing WASM Binary Size` section causes build failure with `leptos_meta`

**Describe the bug**
In the [Appendix: Optimizing WASM Binary Size](https://leptos-rs.github.io/leptos/appendix_binary_size.html#appendix-optimizing-wasm-binary-size) section it says to add this to strip panic code from std
```toml
[unstable]
build-std = ["std", "panic_abort", "core", "alloc"]
build-std-features = ["panic_immediate_abort"]
```
This of course works, but it may cause the following error with `index_map` when using `leptos_meta`
```rust
error[E0107]: struct takes 3 generic arguments but 2 generic arguments were supplied
   --> \cargo\registry\src\index.crates.io-6f17d22bba15001f\leptos_meta-0.3.1\src\lib.rs:102:13
    |
102 |             IndexMap<
    |             ^^^^^^^^ expected 3 generic arguments
103 |                 Cow<'static, str>,
    |                 -----------------
104 |                 (HtmlElement<AnyElement>, Scope, Option<web_sys::Element>),
    |                 ---------------------------------------------------------- supplied 2 generic arguments
    |
note: struct defined here, with 3 generic parameters: `K`, `V`, `S`
   --> \cargo\registry\src\index.crates.io-6f17d22bba15001f\indexmap-1.9.3\src\map.rs:76:12
    |
76  | pub struct IndexMap<K, V, S> {
    |            ^^^^^^^^ -  -  -
help: add missing generic argument
    |
104 |                 (HtmlElement<AnyElement>, Scope, Option<web_sys::Element>), S,
    |                                                                           +++

For more information about this error, try `rustc --explain E0107`.
```
Going to the code in question, we see this. So it's clear on why it's happening. The std rebuild is causing it to hit `not(has_std))`
```rust
70 #[cfg(has_std)]
71 pub struct IndexMap<K, V, S = RandomState> {
72     pub(crate) core: IndexMapCore<K, V>,
73     hash_builder: S,
74 }
75 #[cfg(not(has_std))]
76 pub struct IndexMap<K, V, S> {
77     pub(crate) core: IndexMapCore<K, V>,
78     hash_builder: S,
79 }
```

In my particular case, the error only showed up when I was using `rust-toolchain.toml` set to `channel = "nightly-2023-06-15"`, no idea why, because it works 100% with the SAME toolchain if I don't have `rust-toolchain.toml` in my project.

This problem can be fixed by also adding the following configuration to `config.toml`
```toml
[build]
rustflags = ["--cfg=has_std"]
```

--- a/docs/book/src/appendix_binary_size.md
+++ b/docs/book/src/appendix_binary_size.md
@@ -47,6 +47,12 @@ Note that if you're using this with SSR too, the same Cargo profile will be appl
 target = "x86_64-unknown-linux-gnu" # or whatever
 ```
 
+Also note that in some cases, the cfg feature `has_std` will not be set, which may cause build errors with some dependencies which check for `has_std`. You may fix any build errors due to this by adding:
+```toml
+[build]
+rustflags = ["--cfg=has_std"]
+```
+
 And you'll need to add `panic = "abort"` to `[profile.release]` in `Cargo.toml`. Note that this applies the same `build-std` and panic settings to your server binary, which may not be desirable. Some further exploration is probably needed here.
 
 5. One of the sources of binary size in WASM binaries can be `serde` serialization/deserialization code. Leptos uses `serde` by default to serialize and deserialize resources created with `create_resource`. You might try experimenting with the `miniserde` and `serde-lite` features, which allow you to use those crates for serialization and deserialization instead; each only implements a subset of `serde`’s functionality, but typically optimizes for size over speed.

